tests/operator_test.go

```diff
@@ -526,40 +526,54 @@ var _ = Describe("[Serial][sig-operator]Operator", Serial, func() {
 			return
 		}
 
-		parseImage = func(name, image string) (registry, imagePrefix, version string) {
-			imageRegEx := regexp.MustCompile(fmt.Sprintf("%s%s%s", `^(.*)/(.*)`, name, `([@:].*)?$`))
+		parseImage = func(image string) (registry, imageName, version string) {
+			var getVersion func(matches [][]string) string
+			var imageRegEx *regexp.Regexp
+			var expectedMatchLen int
+
+			if strings.Contains(image, "@sha") {
+				imageRegEx = regexp.MustCompile(`^(.*)/(.*)(@sha.*:)(.*)$`)
```

Comment:
```suggestion
				imageRegEx = regexp.MustCompile(`^(.+)/(.+)(@sha\d+:)([\da-fA-F]+)$`)
```
- `*` matches 0 character, which we probably don't want here
- sha types are always numerical values AFAIK
- shas are always hex

Reply:
Thanks a lot! Makes total sense!